src/Compilers/CSharp/csc/Program.cs

```diff
@@ -33,8 +33,9 @@ private static int MainCore(string[] args)
             ExitingTraceListener.Install();
 #endif
 
-            using var logger = new CompilerServerLogger();
-            return BuildClient.Run(args, RequestLanguage.CSharpCompile, Csc.Run, logger);
+            var requestId = Guid.NewGuid();
```

Comment:
It seems like the only time we aren't generating a new request ID per logger is in VBCSCompiler, but it doensn't seem to call any of the log-with-id methods anyway. 

Would it make more sense to just have the logger make a new Guid when you create it, and have it append to the log messages, rather than making the clients of the logger responsible for tracking it?

Reply:
There are two types of IDs now:

- ID of the instance of CompilerServer logger. This is generally a named instance which makes it easy for us to know which process is writing the log entry. 
- ID of an individual compiler request. This is generated at the MSBuild and command line compiler layer. This ID allows us to track a request, by simple text search in the log, as the request pings between the client and compiler server. Particularly it lets us track back a failure to the original request and lets us see where exactly the request failed.